blog-content/web-socket-recording.md

```diff
@@ -0,0 +1,45 @@
+---
+title: 'WebSocket Recording'
+createdAt: 2023-07-17T15:57:12.147Z
+readingTime: 5
+authorFirstName: Spencer
+authorLastName: Amarantides
+authorTitle: Software Engineer @ Highlight
+authorTwitter: 'https://twitter.com/SpennyNDaJets'
+authorLinkedIn: 'https://www.linkedin.com/in/spencer-amarantides/'
+authorGithub: 'https://github.com/SpennyNDaJets'
+authorWebsite: ''
+authorPFP: 'https://lh3.google.com/u/0/ogw/AGvuzYYcxLXVI9fxSgSPXbQoBw2mpUKll1Kt5hR3LxEh=s64-c-mo'
+tags: Developer Tooling, Launch Week 2
+metaTitle: Session replay now supports recording WebSocket events
+---
+
+Highlight's session replay feature now supports recording WebSocket events! When viewing a session, you were able to view the network requests and their responses in the Developer Tools. However, any communications using WebSockets were missing from the record. Starting in version `7.3.0`, Highlight will record your WebSocket events and display them to help you debug what may be happening behind the scenes. In this blog post, I want to share some of the decisions we may in supporting WebSocket recording and how you can start using it.
+
+![New WebSocket recording feature](/images/blog/web-socket-recording/overview.png)
+
+## Frontend UI/UX
+
+We were between two options for how to display WebSocket messages to our users:
+<ol>
+  <li>Display messages intertwined with other network requests</li>
+  <li>Pull out the messages to their own panel</li>
+</ol>
+
+The first option would allow us to better see the relationship between WebSocket messages and other HTTP requests, as they would be in the same table by chronological order. However, the big concern is the WebSocket messages would inundate the table, making it hard to decipher which messages were important. As a result, the WebSocket's open connection request is placed in the request table with the other HTTP requests, but the individual messages related to a WebSocket are pulled out into their own chronological table. While the messages are not intertwined, they will still include a timestamp that can be used to related to the other requests.
```

Comment:
```suggestion
The first option would allow us to better see the relationship between WebSocket messages and other HTTP requests, as they would be in the same table by chronological order. However, the big concern with that approach was that the WebSocket messages would inundate the table, making it hard to decipher which messages were important. As a result, we decided to put the WebSocket's open connection request in the request table with the other HTTP requests, and keep the individual messages related to a WebSocket pulled out into their own chronological table. While the messages are not intertwined, they will still include a timestamp that can be used to related to the other requests.
```